core/storage/collection/gae_models.py

```diff
@@ -202,6 +202,13 @@ def has_reference_to_user_id(cls, user_id):
         Returns:
             bool. Whether any models refer to the given user ID.
         """
+        for snapshot_content_model in cls.SNAPSHOT_CONTENT_CLASS.get_all():
```

Comment:
Won't get_all() be quite an expensive operation? (especially for explorations?)

Reply:
A query might be better instead (if we can make the fields queryable first).

Reply:
It is not really easy to make the fields we want queryable, this would need a rethinking of the whole `VersionedModel` and its snapshots concept. Since these methods are gonna be used just in the user ID migration for now, we can think about the different approach regarding snapshots later.

Reply:
Done.